src/main/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImpl.java

```diff
@@ -83,13 +80,33 @@ public void buildAndDeploy() throws InterruptedException {
 		buildAndDeployStack(context, sharedResourceStackName, TEMPALTE_SHARED_RESOUCES_MAIN_JSON_VTP, sharedParameters);
 		// Wait for the shared resources to complete
 		Stack sharedStackResults = cloudFormationClient.waitForStackToComplete(sharedResourceStackName);
-		
+
+		createReadOnlyUsers(sharedStackResults);
+
 		// Build each bean stalk environment.
 		List<String> environmentNames = buildEnvironments(sharedStackResults);
 		// setup a web-ACL for each environment
 		aclBuilder.buildWebACL(environmentNames);
 	}
 
+	private void createReadOnlyUsers(Stack sharedStackResults)  {
+		List<String> dbEndpoints = extractDatabaseEndpoints(sharedStackResults);
+		String readOnlyUser = config.getProperty(PROPERTY_KEY_STACK)+config.getProperty(PROPERTY_KEY_INSTANCE)+"readonlyuser";
+		String readOnlyPassword = config.getProperty(PROPERTY_KEY_READONLYUSER_PASSWORD);
+		String schema = config.getProperty(PROPERTY_KEY_STACK)+config.getProperty(PROPERTY_KEY_INSTANCE);
+		String rootUser = config.getProperty(PROPERTY_KEY_STACK)+config.getProperty(PROPERTY_KEY_INSTANCE)+"user";
+		String rootPwd = secretBuilder.getRepositoryDatabasePassword();
+		try {
```

Comment:
You might want to push the try catch inside the loop so that the exception is more meaningful (e.g. you could add the endpoint)